spec/acceptance/puppetserver_upgrade_spec.rb

```diff
@@ -13,29 +13,43 @@
     raise 'At least 256MB free memory required' if memoryfree_mb < 256
   end
 
+  case ENV['BEAKER_PUPPET_COLLECTION']
+  when 'puppet7'
+    from_version = '7.0.0'
+    to_version = '7.2.0'
+  when 'puppet6'
+    from_version = '6.7.0'
+    to_version = '6.7.2'
+  when 'puppet5'
+    from_version = '5.3.6'
+    to_version = '5.3.7'
+  else
+    raise 'Unsupported Puppet collection'
+  end
+
   case fact('osfamily')
   when 'Debian'
-    from_version = "5.3.6-1#{fact('lsbdistcodename')}"
-    to_version = "5.3.7-1#{fact('lsbdistcodename')}"
+    from_version_exact = "6.7.0-1#{fact('lsbdistcodename')}"
+    to_version_exact = "6.7.2-1#{fact('lsbdistcodename')}"
```

Comment:
@ekohl Is this not the reason it is finding puppet6 on Debian? As this code says, when its Debian always use these versions.

Reply:
That explains it. I'm not sure how I missed that when writing it.